objectivec/Tests/GPBCodedInputStreamTests.m

```diff
@@ -62,31 +62,28 @@ - (NSData*)bytes_with_sentinel:(int32_t)unused, ... {
 #define bytes(...) [self bytes_with_sentinel:0, __VA_ARGS__, 256]
 
 - (void)testDecodeZigZag {
-  XCTAssertEqual(0, GPBDecodeZigZag32(0));
-  XCTAssertEqual(-1, GPBDecodeZigZag32(1));
-  XCTAssertEqual(1, GPBDecodeZigZag32(2));
-  XCTAssertEqual(-2, GPBDecodeZigZag32(3));
-  XCTAssertEqual((int32_t)0x3FFFFFFF, GPBDecodeZigZag32(0x7FFFFFFE));
-  XCTAssertEqual((int32_t)0xC0000000, GPBDecodeZigZag32(0x7FFFFFFF));
-  XCTAssertEqual((int32_t)0x7FFFFFFF, GPBDecodeZigZag32(0xFFFFFFFE));
-  XCTAssertEqual((int32_t)0x80000000, GPBDecodeZigZag32(0xFFFFFFFF));
-
-  XCTAssertEqual((int64_t)0, GPBDecodeZigZag64(0));
-  XCTAssertEqual((int64_t)-1, GPBDecodeZigZag64(1));
-  XCTAssertEqual((int64_t)1, GPBDecodeZigZag64(2));
-  XCTAssertEqual((int64_t)-2, GPBDecodeZigZag64(3));
-  XCTAssertEqual((int64_t)0x000000003FFFFFFFL,
-                 GPBDecodeZigZag64(0x000000007FFFFFFEL));
-  XCTAssertEqual((int64_t)0xFFFFFFFFC0000000L,
-                 GPBDecodeZigZag64(0x000000007FFFFFFFL));
-  XCTAssertEqual((int64_t)0x000000007FFFFFFFL,
-                 GPBDecodeZigZag64(0x00000000FFFFFFFEL));
-  XCTAssertEqual((int64_t)0xFFFFFFFF80000000L,
-                 GPBDecodeZigZag64(0x00000000FFFFFFFFL));
-  XCTAssertEqual((int64_t)0x7FFFFFFFFFFFFFFFL,
-                 GPBDecodeZigZag64(0xFFFFFFFFFFFFFFFEL));
-  XCTAssertEqual((int64_t)0x8000000000000000L,
-                 GPBDecodeZigZag64(0xFFFFFFFFFFFFFFFFL));
+  [self assertReadZigZag32:bytes(0x0) value:0];
+  [self assertReadZigZag32:bytes(0x1) value:-1];
+  [self assertReadZigZag32:bytes(0x2) value:1];
+  [self assertReadZigZag32:bytes(0x3) value:-2];
+
+  [self assertReadZigZag32:bytes(0xFE, 0xFF, 0xFF, 0xFF, 0x07) value:(int32_t)0x3FFFFFFF];
+  [self assertReadZigZag32:bytes(0xFF, 0xFF, 0xFF, 0xFF, 0x07) value:(int32_t)0xC0000000];
+  [self assertReadZigZag32:bytes(0xFE, 0xFF, 0xFF, 0xFF, 0x0F) value:(int32_t)0x7FFFFFFF];
+  [self assertReadZigZag32:bytes(0xFF, 0xFF, 0xFF, 0xFF, 0x0F) value:(int32_t)0x80000000];
+
+  [self assertReadZigZag64:bytes(0x0) value:0];
+  [self assertReadZigZag64:bytes(0x1) value:-1];
+  [self assertReadZigZag64:bytes(0x2) value:1];
+  [self assertReadZigZag64:bytes(0x3) value:-2];
+
+  [self assertReadZigZag64:bytes(0xFE, 0xFF, 0xFF, 0xFF, 0x07) value:(int32_t)0x3FFFFFFF];
+  [self assertReadZigZag64:bytes(0xFF, 0xFF, 0xFF, 0xFF, 0x07) value:(int32_t)0xC0000000];
+  [self assertReadZigZag64:bytes(0xFE, 0xFF, 0xFF, 0xFF, 0x0F) value:(int32_t)0x7FFFFFFF];
+  [self assertReadZigZag64:bytes(0xFF, 0xFF, 0xFF, 0xFF, 0x0F) value:(int32_t)0x80000000];
+
+  [self assertReadZigZag64:bytes(0xFE, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x01) value:0x7FFFFFFFFFFFFFFFL];
+  [self assertReadZigZag64:bytes(0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x01) value:0x8000000000000000L];
 }
 
 - (void)assertReadVarint:(NSData*)data value:(int64_t)value {
```

Comment:
Maybe a comment now on this to call out that it only works for values 0-INT32_MAX?  (and the other helpers like this below)

Reply:
updated to check for ranges